### ISSUE TITLE
CI fails non-deterministically

### ISSUE THREAD

**Issue by leungbk** [emacs-lsp/lsp-mode#2686]
After merging https://github.com/emacs-lsp/lsp-mode/commit/b77aecf2d613c7ed5cadb9c5c37fcbb205c3f3a9, I noticed several different CI failures across different runs:

1. Missing dependencies.

> cask install
> Some dependencies were not available: lv, markdown-mode, ht, f, deferred, undercover, ecukes, espuds, ert-runner

2. Out-of-memory error. I noticed this on Windows with an earlier run of the same commit.

It would be nice if we could remove any non-determinism.

**Comment by ericdallo** [comment]
Have no idea why this is happening, any clues @jcs090218 ?

**Comment by jcs090218** [comment]
> Have no idea why this is happening, any clues @jcs090218 ?

No, I have no clue on this. Let me report to cask and see what they say.

**Comment by yyoncho** [comment]
> After merging [b77aecf](https://github.com/emacs-lsp/lsp-mode/commit/b77aecf2d613c7ed5cadb9c5c37fcbb205c3f3a9), I noticed several different CI failures across different runs:

I am seeing these errors from time to time. Not sure if they are related to b77aecf . Maybe we should consider using straight for the CI? Although, at least for now the "official" way to use lsp-mode is using package.el.

**Comment by conao3** [comment]
Sorry for your inconvenience on cask. I've released cask v0.8.6. please update it?

**Comment by conao3** [comment]
Some comment on my side, `cask` just install using `package-install`. so if package install failed, the server is the main culprit. In particular, ELPA is dropping randomly.
You can increase its reliability by hosting a separate server that only hosts the lsp-mode dependency.
This is very easy, see my [CELPA](https://celpa.conao3.com/) and its [repository](https://github.com/conao3/celpa)

**Comment by kiennq** [comment]
> Maybe we should consider using straight for the CI? Although, at least for now the "official" way to use lsp-mode is using package.el.

JFYI, using `quelpa` is closer to `package.el` than `straight.el`.

**Comment by yyoncho** [comment]
> JFYI, using `quelpa` is closer to `package.el` than `straight.el`.

What do you mean?

**Comment by kiennq** [comment]
> > JFYI, using `quelpa` is closer to `package.el` than `straight.el`.
> 
> What do you mean?

`Quelpa` is using `package.el` to install package, so it's will be closer experience compare to `straight.el`, which is doing byte-compiling and package placement by itself.

**Comment by jcs090218** [comment]
> JFYI, using quelpa is closer to package.el than straight.el.

I agree with @kiennq. Quelpa seems to be a lot stabler than regular `package.el`.
